devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,518 @@
+        : IPoseSensor(aName, aStatus)
+        , m_frameTransformHandler(frameTransformHandler)
+    {}
+
+    ~PoseSensor() override = default;
+
+    // -------------------------
+    // IOrientationSensor interface
+    // -------------------------
+    bool getPose(Quaternion& orientation, Vector3& position) const override
+    {
+        return this->m_frameTransformHandler->getTransform(orientation, position);
+    }
+
+    // ------------------------
+    // Custom utility functions
+    // ------------------------
+    inline void setStatus(const wearable::sensor::SensorStatus aStatus) { m_status = aStatus; }
+
+private:
+    IFrameTransformHandler* m_frameTransformHandler = nullptr;
+};
+
+// TODO: implement other sensors
+
+// ======================
+// IFrameTransformToIWear
+// ======================
+
+wearable::sensor::SensorType sensorTypeFromString(std::string sensorTypeString)
+{
+    if (sensorTypeString == "Accelerometer")
+        return wearable::sensor::SensorType::Accelerometer;
+    else if (sensorTypeString == "EmgSensor")
+        return wearable::sensor::SensorType::EmgSensor;
+    else if (sensorTypeString == "Force3DSensor")
+        return wearable::sensor::SensorType::Force3DSensor;
+    else if (sensorTypeString == "ForceTorque6DSensor")
+        return wearable::sensor::SensorType::ForceTorque6DSensor;
+    else if (sensorTypeString == "FreeBodyAccelerationSensor")
+        return wearable::sensor::SensorType::FreeBodyAccelerationSensor;
+    else if (sensorTypeString == "Gyroscope")
+        return wearable::sensor::SensorType::Gyroscope;
+    else if (sensorTypeString == "Magnetometer")
+        return wearable::sensor::SensorType::Magnetometer;
+    else if (sensorTypeString == "OrientationSensor")
+        return wearable::sensor::SensorType::OrientationSensor;
+    else if (sensorTypeString == "PoseSensor")
+        return wearable::sensor::SensorType::PoseSensor;
+    else if (sensorTypeString == "PositionSensor")
+        return wearable::sensor::SensorType::PositionSensor;
+    else if (sensorTypeString == "SkinSensor")
+        return wearable::sensor::SensorType::SkinSensor;
+    else if (sensorTypeString == "TemperatureSensor")
+        return wearable::sensor::SensorType::TemperatureSensor;
+    else if (sensorTypeString == "Torque3DSensor")
+        return wearable::sensor::SensorType::Torque3DSensor;
+    else if (sensorTypeString == "VirtualLinkKinSensor")
+        return wearable::sensor::SensorType::VirtualLinkKinSensor;
+    else if (sensorTypeString == "VirtualJointKinSensor")
+        return wearable::sensor::SensorType::VirtualJointKinSensor;
+    else if (sensorTypeString == "VirtualSphericalJointKinSensor")
+        return wearable::sensor::SensorType::VirtualSphericalJointKinSensor;
+    else {
+        yError() << LogPrefix << "Sensor type" << sensorTypeString << "not supported";
+        return {};
```

Comment:
I don't remember, do we have an invalid sensor type?

Reply:
I don't think. See https://github.com/robotology/wearables/blob/c6a8478343881468b91112244bd25c6a803d456e/interfaces/IWear/include/Wearable/IWear/IWear.h#L75-L92

Reply:
Do you think that adding a new `sensor::SensorType::Invalid` here would create any problems in the existing devices?

Reply:
I was looking around don't see any issue, but we should update all the `getSensor` methods in the devices (e.g. https://github.com/robotology/wearables/blob/master/devices/IWearRemapper/src/IWearRemapper.cpp#L862-L948) to consider this case. If you think it is worth I can handle it.

Reply:
I have added the `invalid` type in https://github.com/robotology/wearables/pull/126/commits/7bbce5d0648e499d5e29ad85af6c705b5ae02ec2. However, I didn't changed the other device to take in consideration this case, I will go through them in order to understand which change is needed. Concerning the [thrift message](https://github.com/robotology/wearables/blob/master/msgs/thrift/WearableData.thrift#L173-L191) I am not sure if it should be added since we will not have an associated message.

Reply:
For the time being I have also not included the `Invalid` sensor type in the list of `AllSensorsType` [here](https://github.com/robotology/wearables/blob/add-iFrameTransformToIWear-device/interfaces/IWear/include/Wearable/IWear/IWear.h#L75-L102). That is because I think invalid sensors can be discarded when getting all the sensors from a device (I think that is where those lists are used).
@diegoferigo What do you think? I have to say for the time being I am struggling a bit to find a useful usage of the `Invalid` type, but there might be some corner case in which it will turn out to be useful

Reply:
Sorry for the late reply @lrapetti. Thanks for iterating on this.

> I have added the invalid type in 7bbce5d. However, I didn't changed the other device to take in consideration this case, I will go through them in order to understand which change is needed.

There should not be major problems in my opinion, worst case there could be a compiler warning if the `default` case is not used (and this should be the desired behavior because it is an easy way to spot where new cases should be added).

> Concerning the thrift message I am not sure if it should be added since we will not have an associated message.

This is more tricky. I'm not sure what could be the effect, and I suspect that any modification of the thrift message could make previous datasets incompatible. Maybe, considering that the addition is kind-of backward compatible, it is not the case, but we should keep this in mind and assess if it's worth.

> For the time being I have also not included the `Invalid` sensor type in the list of `AllSensorsType` [here](https://github.com/robotology/wearables/blob/add-iFrameTransformToIWear-device/interfaces/IWear/include/Wearable/IWear/IWear.h#L75-L102). That is because I think invalid sensors can be discarded when getting all the sensors from a device (I think that is where those lists are used).
> @diegoferigo What do you think? I have to say for the time being I am struggling a bit to find a useful usage of the `Invalid` type, but there might be some corner case in which it will turn out to be useful

Over time I lost a bit the overall picture of wearable and hde, I fear that my opinion could miss something important. From your comment, it seems reasonable to just add the valid sensor types to `AllSensorsType`.

Reply:
mutex added in https://github.com/robotology/wearables/pull/126/commits/faf2bbce350a99f7ecc78983839a932fc5d977be

Reply:
> > Concerning the thrift message I am not sure if it should be added since we will not have an associated message.
> 
> This is more tricky. I'm not sure what could be the effect, and I suspect that any modification of the thrift message could make previous datasets incompatible. Maybe, considering that the addition is kind-of backward compatible, it is not the case, but we should keep this in mind and assess if it's worth.

Yes, adding this type to the message will make all the current dataset no longer compatible, indeed I would not consider adding it for the time being.